### PR TITLE
fix logistic regression problem.

### PR DESCRIPTION
**Pull request by mcreixell**
By reshaping the data as we discussed in #315, logistic regression doesn't seem to work... I also tried SVC and showed similar performance. 

image

### CODE REVIEW

**Comment by mcreixell** [comment]
EGFRm/ALKf and especially STK11 seem to work better than TP53 after this fix. Do you think this prediction is good enough to use? Even so interpretation of EGFR/ALK wouldn't be easy since there are so many "relevant" clusters. 

image

image

**Comment by mcreixell** [comment]
This last commit fixes the same problem to predict immune infiltration. With this new set up, maybe we're able to predict mutations we couldn't before. I'm going to explore this next here.

**Comment by mcreixell** [comment]
With this new set up, maybe we're able to predict mutations we couldn't before in `FigureM4.svg.` I'm going to explore this next here.

**Comment by mcreixell** [comment]
SVC seems to work better than LR with our method at predicting mutations while showing the same performance at predicting sample type (NAT vs tumor) so I switched them in all regressions.

**Comment by aarmey** [comment]
Hmm... LinearSVC should have essentially identical performance to logistic regression. Also, an SVM comes with the downside that you can't see which clusters it depends on. Is the difference really that big?

**Comment by mcreixell** [comment]
Looking at the ROC STK11 plots, I think SVC works slightly better than LASSO and much better than elastic net at predicting both genotypes. However, I didn't pay attention before to the fact that the cluster weights between SVC and LASSO are basically the same, presumably because I'm setting the `penalty` parameter to `l1`. @aarmey 

image

image

image

Similar conclusions with EGFR:

image

image

image

Predicting EGFRm/ALKf works better but not drastically, maybe it'd be more interesting for interpretation purposes to keep EGFR alone? 

image

**Comment by aarmey** [comment]
SVC does not give you cluster weights—you are still plotting logistic regression model weights. In SVC you can only see the support vectors, or the training points that matter, not the variables.

**Comment by mcreixell** [comment]
So LinearSVC’s attribute` .coef_` calculates logistic regression coefficients? When running SVC I’m not using LogisticRegressionCV at all.

**Comment by mcreixell** [comment]
I switched back to LR and played with the arguments. These are the best results I could get.

File: msresist/figures/figureM3.py [hunk 1]
```diff
@@ -53,19 +53,18 @@ def makeFigure():
     tt = centers.iloc[:, -1]
     tt = tt.replace("Normal", 0)
     tt = tt.replace("Tumor", 1)
-    lr = LogisticRegressionCV(cv=4, solver="saga", max_iter=10000, n_jobs=-1, penalty="elasticnet", class_weight="balanced", l1_ratios=[0.5, 0.9]).fit(c, tt)
+    svc = LinearSVC(penalty="l1", dual=False, max_iter=10000, tol=1e-7)
```
Comment (aarmey) [inline msresist/figures/figureM3.py:56]: You're still using LinearSVC here?

**Review by aarmey — CHANGES_REQUESTED** [review]
Looks good. Just one question.

File: msresist/figures/figureM3.py (hunk 1 above)
Comment (mcreixell) [reply msresist/figures/figureM3.py]: Oops forgot to change this one. Solved below.

**Review by aarmey — APPROVED** [review]

**Pull request merged**